### ISSUE TITLE
Bumping 4.82 to 4.84 causes recreation of secret version

### ISSUE THREAD

**Issue by AtzeDeVries** [hashicorp/terraform-provider-google#16611]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

1.3.6

### Affected Resource(s)

* google_secret_manager_secret_version

### Terraform Configuration Files

```tf
  # redacted must be replaced
resource "google_secret_manager_secret_version" "redacted" {
  secret      = google_secret_manager_secret.redacted.id
  secret_data = base64decode(redacted)
}
```

### Terraform Configuration Files

```tf
  # redacted must be replaced
-/+ resource "google_secret_manager_secret_version" "redacted" {
      ~ create_time           = "XX-XX-XXT08:01:15.814929Z" -> (known after apply)
      + deletion_policy       = "DELETE"
      ~ destroy_time          = "" -> (known after apply)
      ~ id                    = "redactedversions//1" -> (known after apply)
      + is_secret_data_base64 = false # forces replacement
      ~ name                  = "redacted/versions/1" -> (known after apply)
      ~ version               = "1" -> (known after apply)
        # (3 unchanged attributes hidden)
    }
```

### References

https://github.com/hashicorp/terraform-provider-google/releases/tag/v4.84.0 introduces the variable `is_secret_data_base64`

**Comment by edwardmedia** [comment]
@AtzeDeVries what do you see if bumping to v5.x?

**Comment by AtzeDeVries** [comment]
Hi @edwardmedia ,

Thanks for the reply. I'll try to do that, but in our environment it is a bit complicated to just test on the current state files we have. I am preparing for it. I have been reading a bit of code, please take a look at these examples which kinda shows (according the code) the intent to recreate

Shows intent to create the value
https://github.com/hashicorp/terraform-provider-google/blob/5e1c8e2b799fbb4de246b9af17d4628d5a5ec641/google/services/secretmanager/resource_secret_manager_secret_version.go#L227-L232

Shows that creating the value recreates the secret
https://github.com/hashicorp/terraform-provider-google/blob/5e1c8e2b799fbb4de246b9af17d4628d5a5ec641/google/services/secretmanager/resource_secret_manager_secret_version.go#L119-L125

If i view a object in state, the field `is_secret_data_base64` does not exist (`terraform show 'resourceid')

```
# module.advanced-analytics-website-data-gcp-project[0].google_secret_manager_secret_version.privatekey_data["datadog"]:
resource "google_secret_manager_secret_version" "redact" {
    create_time  = "2022-02-11T12:44:18.073553Z"
    destroy_time = ""
    enabled      = true
    id           = "projects/redact/versions/1"
    name         = "projects/redact"
    secret       = "projects/redact"
    secret_data  = (sensitive value)
    version      = "1"
}
```

**Comment by edwardmedia** [comment]
@AtzeDeVries yeah, **is_secret_data_base64** is not in the state and it is optional, I can't explain why below line in your plan. 

```tf
+ is_secret_data_base64 = false # forces replacement
```

Again, using my config [above](https://github.com/hashicorp/terraform-provider-google/issues/16611#issuecomment-1830255943), it appears working as expected.

**Comment by AtzeDeVries** [comment]
I cannot reproduce local either. I now tried to bump to 4.83 which added the deletion policy to the state. Bumping from 4.83 to .84 resolved the issue. I did not check why and at the moment not willing to spend time on it ;) . @edwardmedia  if you dont want to research it please close the issue.

**Comment by edwardmedia** [comment]
Let's close this right now. Please reopen the issue if you do it again.
